### PR TITLE
Cleanup cloudcare

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @dimagi/js-team and the touchforms portion. unfortunately the only real way to read this is commit-by-commit. had to do it very piecemeal

screen shot 2015-12-06 at 12 29 16 am and i added tests!

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 36 errors:

* touchforms/formplayer/static/formplayer/script/utils.js, line 3 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 12 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 17 - Unnecessary semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 52 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 54 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 99 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 100 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 107 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 131 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 140 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 152 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 153 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 154 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 185 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 192 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 218 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 237 - Use '===' to compare with 'null'.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 244 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 252 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 291 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 305 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 315 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 335 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 355 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 358 - ['type'] is better written in dot notation.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 360 - ['type'] is better written in dot notation.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 361 - ['reason'] is better written in dot notation.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 363 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 373 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 88 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 100 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 128 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 147 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 168 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 176 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 183 - Missing semicolon.

**Comment by benrudolph** [comment]
actually i suppose it's not too bad to just view the diff

**Comment by dimagimon** [comment]
There are 27 errors:

* touchforms/formplayer/static/formplayer/script/utils.js, line 3 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 12 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 17 - Unnecessary semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 52 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 54 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 112 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 124 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 133 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 145 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 146 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 147 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 183 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 190 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 219 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 233 - Use '===' to compare with 'null'.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 240 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 248 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 287 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 301 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 311 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 331 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 351 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 359 - Missing semicolon.
* touchforms/formplayer/static/formplayer/script/webformsession.js, line 369 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 90 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 167 - Missing semicolon.
* touchforms/spec/javascripts/webformsession_spec.js, line 175 - Missing semicolon.

**Comment by snopoke** [comment]
This is great! Nice work.

File: touchforms/formplayer/static/formplayer/script/webformsession.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    };
+    accumulate_answers(form);
+
+    this.serverRequest({
+            'action': Formplayer.Const.SUBMIT,
+            'answers': answers,
+            'prevalidated': prevalidated
+        },
+        function(resp) {
+            if (resp.status == 'success') {
```
Comment (orangejenny) [inline touchforms/formplayer/static/formplayer/script/webformsession.js]: triple equals?

**Comment by orangejenny** [comment]
this looks great

**Comment by czue** [comment]
this is beastly! it looked like several folks had already reviewed so I just took a super quick pass but big :+1: (and awesome to see the tests!)

**Comment by benrudolph** [comment]
:white_check_mark:

**Comment by benrudolph** [comment]
(nvm, still want to sanity check edit forms)

**Comment by benrudolph** [comment]
ok g2g once tests pass on main HQ repo

**Comment by dannyroberts** [comment]
The dark ages are coming to an end

**Comment by czue** [comment]
:tada:

**Comment by benrudolph** [comment]
:sunny: put it on the board! ben 1, tech debt 12832

**Pull request merged**